src/AvaloniaMath/Colors/PredefinedColorParser.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using Avalonia.Media;
+using System.Xml.Linq;
+
+namespace WpfMath.Colors
+{
+    public class PredefinedColorParser : IColorParser
+    {
+        private const string ResourceName = TexUtilities.ResourcesDataDirectory + "PredefinedColors.xml";
+        public static readonly PredefinedColorParser Instance = new PredefinedColorParser(ResourceName);
+
+        private readonly IReadOnlyDictionary<string, Color> _colors;
+
+        private PredefinedColorParser(string resourceName)
+        {
+            using (var resource = Assembly.GetExecutingAssembly().GetManifestResourceStream(resourceName))
+            {
+                var doc = XDocument.Load(resource);
+                _colors = Parse(doc.Root);
+            }
+        }
+
+        public Color? Parse(IEnumerable<string> components)
+        {
+            var componentList = components.ToList();
+            var hasAlphaComponent = componentList.Count == 2;
+            if (componentList.Count != 1 && !hasAlphaComponent)
+                return null;
+
+            var colorName = componentList[0];
+            if (!_colors.TryGetValue(colorName, out var color))
+                return null;
+
+            byte? alpha = 255;
+            if (hasAlphaComponent)
+            {
+                var alphaFraction = double.TryParse(
+                    componentList[1],
+                    NumberStyles.AllowDecimalPoint,
+                    CultureInfo.InvariantCulture,
+                    out var a)
+                    ? (double?) a
+                    : null;
+                if (alphaFraction == null || alphaFraction < 0.0 || alphaFraction > 1.0)
+                    return null;
+
+                alpha = (byte) Math.Round(255.0 * a, MidpointRounding.AwayFromZero);
+            }
+
+            if (alpha == null)
+                return null;
+
+            color = Color.FromArgb(alpha.Value, color.R, color.G, color.B);
```

Comment:
> Copy-paste to Avalonia/Colors from WPF/Colors

Sorry, but I have no idea but to do this. It differs only in this line (without `Avalonia.Media`)

Reply:
You can still try the simple way using preprocessor directives and the fact that the project for avalonia is built under netstandard

Reply:
> because no setters

[readonly struct](https://github.com/AvaloniaUI/Avalonia/commit/d55edaa18a2791ff024491f25820022c1d1a8531#diff-347257ebc1fd2f1d8dd665dec376da17)

Reply:
It won't work via preprocessor, because of how .NET compilation model is organized. If we want to leave _one_ version of the common library, then we'll need to compile it _once_, with exactly _one_ set of the preprocessor definitions.

I'll think about it, probably a bit of generic magic/parametrization will help us here.

Reply:
> If we want to leave _one_ version of the common library, then we'll need to compile it _once_

Sorry, I have no experience, so I did not think about this

Reply:
You know what? Let's leave it as-isto not complicate the merge. Generalization will be performed on another stage of work.